CLI: stat command unit test fails when namespace is not "default"

## Bug Report

### What is the issue?

* When the `kubectl` context is set to use another namespace as the default, unit tests in `stat_test.go` start failing. 
* I think the issue is that `linkerd` for most commands (that act on namespaced resources) now sets the default namespace from the context; when `stat.go` validates `namespace` flags, it makes sure the `--namespace` flag is not used as an option when the target resource is a k8s namespace by asserting the namespace option is the same as a hardcoded "default" value. 
* Since the default namespace here is taken from the context, it'll be different to the hardcoded value in the validation.

### How can it be reproduced?

* First, set the default context namespace to something else, 
e.g `kubectl config set-context minikube --namespace=linkerd` --> this will set the default namespace for the minikube context to linkerd
* Second, run the cli tests: `go test -cover -race -v ./cmd/cli/...`

Stat tests should fail in this instance. If the namespace is set back to "default", it will work.

### Logs, error output, etc
```
=== RUN   TestStat/Returns_an_error_if_--time-window_is_not_more_than_15s
    TestStat/Returns_an_error_if_--time-window_is_not_more_than_15s: stat_test.go:203: Expected error [metrics time window needs to be at least 15s] instead got [--namespace flag is incompatible with namespace resource typee]
--- FAIL: TestStat (0.02s)
    --- FAIL: TestStat/Returns_namespace_stats (0.00s)
    --- FAIL: TestStat/Returns_pod_stats (0.00s)
    --- PASS: TestStat/Returns_trafficsplit_stats (0.01s)
    --- FAIL: TestStat/Returns_namespace_stats_(json) (0.00s)
    --- PASS: TestStat/Returns_trafficsplit_stats_(json) (0.00s)
    --- FAIL: TestStat/Returns_all_namespace_stats (0.00s)
    --- FAIL: TestStat/Returns_all_namespace_stats_(json) (0.00s)
    --- FAIL: TestStat/Returns_TCP_stats (0.00s)
    --- FAIL: TestStat/Returns_an_error_for_named_resource_queries_with_the_--all-namespaces_flag (0.00s)
    --- PASS: TestStat/Rejects_commands_with_both_--to_and_--from_flags (0.00s)
    --- PASS: TestStat/Rejects_commands_with_both_--to-namespace_and_--from-namespace_flags (0.00s)
    --- PASS: TestStat/Rejects_commands_with_both_--all-namespaces_and_--namespace_flags (0.00s)
    --- PASS: TestStat/Rejects_--to-namespace_flag_when_the_target_is_a_namespace (0.00s)
    --- PASS: TestStat/Rejects_--from-namespace_flag_when_the_target_is_a_namespace (0.00s)
    --- FAIL: TestStat/Returns_an_error_if_--time-window_is_not_more_than_15s (0.00s)
```

```
CURRENT   NAME                          CLUSTER                                     AUTHINFO                                    NAMESPACE
          east                          gke_final-year-research_us-east1-b_east-1   gke_final-year-research_us-east1-b_east-1   linkerd-multicluster
          kubernetes-admin@kubernetes   kubernetes                                  kubernetes-admin                            
          minikube                      minikube                                    minikube                                    
*         west                          gke_final-year-research_us-west1-a_west-1   gke_final-year-research_us-west1-a_west-1   linkerd-multicluster
```



#### `linkerd check` output

```text
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-identity
----------------
√ certificate config is valid
√ trust anchors are using supported crypto algorithm
√ trust anchors are within their validity period
√ trust anchors are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust anchor

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ tap api service is running

linkerd-version
---------------
√ can determine the latest version
‼ cli is up-to-date
    is running version 20.6.4 but the latest edge version is 20.7.3
    see https://linkerd.io/checks/#l5d-version-cli for hints

control-plane-version
---------------------
‼ control plane is up-to-date
    unsupported version channel: git-f4f62e79
    see https://linkerd.io/checks/#l5d-version-control for hints
‼ control plane and cli versions match
    control plane running git-f4f62e79 but cli running edge-20.6.4
    see https://linkerd.io/checks/#l5d-version-control for hints

linkerd-addons
--------------
√ 'linkerd-config-addons' config map exists

linkerd-grafana
---------------
√ grafana add-on service account exists
√ grafana add-on config map exists
√ grafana pod is running
```

### Environment

- Kubernetes Version: `v1.18.0`
- Cluster Environment: `GKE/Minikube`
- Host OS: `OSX Catalina (10.15.5)`
- Linkerd version: `edge-20.6.4`

### Possible solution

* I think the easiest thing to do here is to replace the hardcoded "default" values with the `defaultNamespace` inferred from the context

### Additional context

Pull requests to see what has been done:
* PR #4512 
* PR #4291 

Original issue:
* Issue #4197

--- a/cli/cmd/stat.go
+++ b/cli/cmd/stat.go
@@ -742,7 +742,7 @@ func (o *statOptions) validateConflictingFlags() error {
 		return fmt.Errorf("--to-namespace and --from-namespace flags are mutually exclusive")
 	}
 
-	if o.allNamespaces && o.namespace != "default" {
+	if o.allNamespaces && o.namespace != defaultNamespace {
 		return fmt.Errorf("--all-namespaces and --namespace flags are mutually exclusive")
 	}
 
@@ -760,9 +760,9 @@ func (o *statOptions) validateNamespaceFlags() error {
 		return fmt.Errorf("--from-namespace flag is incompatible with namespace resource type")
 	}
 
-	// Note: technically, this allows you to say `stat ns --namespace default`, but that
+	// Note: technically, this allows you to say `stat ns --namespace <default-namespace-from-kubectl-context>`, but that
 	// seems like an edge case.
-	if o.namespace != "default" {
+	if o.namespace != defaultNamespace {
 		return fmt.Errorf("--namespace flag is incompatible with namespace resource type")
 	}
 

